lib/cmxl/statement.rb

```diff
@@ -26,12 +26,22 @@ def transactions
     def parse!
       self.fields = []
 
-      # split transactions, each transaction starts with a colon after a linebreak
+      # split transactions, each transaction starts with a colon, 2-3 letters and/or digits, followed by a colon.
       # do not remove line breaks within transaction lines as they are used to determine field details
       # e.g. :61:-supplementary
-      source.split("\n:").each(&:strip!).each do |line|
-        line = ":#{line}" unless line =~ /^:/ # prepend lost : via split
+      matches = []
+      remaining_source = source
+
+      # this code builds an array in matches, [':61:', 'line_content', ':28C', 'more line content']
+      while remaining_source && (match = remaining_source.split(/^(:[A-Z0-9]{2,3}:)(.*)/m))
+        matches << match[0] if !match[0].nil? && !match[0].empty? # this is the remaining line after a potential tag
+        matches << match[1] if !match[1].nil? # this is the tag
+        remaining_source = match[2] # this is everything after the tag
+      end
```

Comment:
I like the approach. But I think it would be easier to understand what's happening if we had an array with sub-arrays containing tag as the first element and the line_content as a second. this would spare the the `each_slice(2)` iteration

Reply:
after digging into your solution and understanding what it does, it's actually quiet clever :)

I've been able to shrink it down to
```ruby
def parse!
  self.fields = []

  lines = source.split(/(:[0-9A-Z]{2,3}:)+/m).reject(&:empty?).each_slice(2).map(&:join)

  lines.map do |line|
    if line =~ /\A:86:/
      if field = fields.last
        field.add_meta_data(line)
      end
    else
      field = Field.parse(line)
      fields << field unless field.nil?
    end
  end
end
```

which seems simple enough for me

Reply:
I like it! Does this make sure the tag is at the beginning of the line, or could a stray `:11:` within the line throw this off?